drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -153,38 +205,59 @@ static int pwm_nrf5_sw_pin_set(const struct device *dev, uint32_t pwm,
 		NRF_GPIO->OUTCLR = BIT(pwm);
 	}
 
-	/* TODO: if the assigned NRF_TIMER supports higher bit resolution,
-	 * use that info in config struct and setup accordingly.
-	 */
-
-	/* calc div, to scale down to fit in 16 bits */
-	div = period_cycles >> 16;
+	/* configure RTC / TIMER */
+	if (USE_RTC) {
+		rtc->EVENTS_COMPARE[channel] = 0;
+		rtc->EVENTS_COMPARE[config->map_size] = 0;
 
-	timer->EVENTS_COMPARE[channel] = 0;
-	timer->EVENTS_COMPARE[config->map_size] = 0;
+		rtc->CC[channel] = pulse_cycles - 1;
+		rtc->CC[config->map_size] = period_cycles - 1;
```

Comment:
Maybe it would be good to add a comment why these `- 1` are needed here.

Reply:
Added a comment about it.

Reply:
Added a comment about it.